[CI] StatelessFieldCapabilitiesIT testFieldCapsForHollowShards failing

**Build Scans:**
- [elasticsearch-intake #39701 / part2](https://gradle-enterprise.elastic.co/s/3wlbuaioorc4q)
- [elasticsearch-pull-request #145505 / part-2](https://gradle-enterprise.elastic.co/s/5mdtyvf64a7ww)
- [elasticsearch-pull-request #144967 / part-2](https://gradle-enterprise.elastic.co/s/bnwrgdn56zbzy)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:internalClusterTest" --tests "org.elasticsearch.xpack.stateless.StatelessFieldCapabilitiesIT.testFieldCapsForHollowShards" -Dtests.seed=3200427929F5C9E5 -Dtests.locale=nds-DE -Dtests.timezone=Etc/GMT+2 -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.stateless.StatelessFieldCapabilitiesIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testFieldCapsForHollowShards'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: an instance of org.elasticsearch.xpack.stateless.engine.IndexEngine
     but: <org.elasticsearch.xpack.stateless.engine.HollowIndexEngine@6c6a0ee4> is a org.elasticsearch.xpack.stateless.engine.HollowIndexEngine
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testFieldCapsForHollowShards
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 3 failures in test testFieldCapsForHollowShards (0.5% fail rate in 657 executions)
- [main] 2 failures in step part-2 (0.5% fail rate in 440 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (0.5% fail rate in 439 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -539,9 +539,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.StatelessCommitServiceIT
   method: testRetainCommitForReadersAfterShardMovedAway
   issue: https://github.com/elastic/elasticsearch/issues/149083
-- class: org.elasticsearch.xpack.stateless.StatelessFieldCapabilitiesIT
-  method: testFieldCapsForHollowShards
-  issue: https://github.com/elastic/elasticsearch/issues/149113
 - class: org.elasticsearch.xpack.esql.qa.csv.CsvFormatSpecIT
   method: test {csv-spec:csv-headerless.multiFileHeaderlessReadSchemaPreventsTypeDrift [LOCAL]}
   issue: https://github.com/elastic/elasticsearch/issues/149126

--- a/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/StatelessFieldCapabilitiesIT.java
+++ b/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/StatelessFieldCapabilitiesIT.java
@@ -11,11 +11,15 @@ import org.elasticsearch.action.NoShardAvailableActionException;
 import org.elasticsearch.action.fieldcaps.FieldCapabilitiesBuilder;
 import org.elasticsearch.action.index.IndexRequest;
 import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.routing.allocation.decider.EnableAllocationDecider.Rebalance;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.core.TimeValue;
 
 import java.util.Map;
 
+import static org.elasticsearch.cluster.InternalClusterInfoService.CLUSTER_ROUTING_ALLOCATION_ESTIMATED_HEAP_THRESHOLD_DECIDER_ENABLED;
+import static org.elasticsearch.cluster.routing.allocation.IndexBalanceConstraintSettings.INDEX_BALANCE_DECIDER_ENABLED_SETTING;
+import static org.elasticsearch.cluster.routing.allocation.decider.EnableAllocationDecider.CLUSTER_ROUTING_REBALANCE_ENABLE_SETTING;
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
 import static org.elasticsearch.xpack.stateless.commits.HollowShardsService.SETTING_HOLLOW_INGESTION_TTL;
 import static org.elasticsearch.xpack.stateless.commits.HollowShardsService.STATELESS_HOLLOW_INDEX_SHARDS_ENABLED;
@@ -56,6 +60,17 @@ public class StatelessFieldCapabilitiesIT extends AbstractStatelessPluginIntegTe
         assertFieldCaps(indexName);
     }
 
+    @Override
+    protected Settings.Builder nodeSettings() {
+        return super.nodeSettings()
+            // disable rebalancing to avoid spurious shard movements that might cause premature hollowing
+            .put(CLUSTER_ROUTING_REBALANCE_ENABLE_SETTING.getKey(), Rebalance.NONE)
+            // disable index balance decider to avoid spurious shard movements that might cause premature hollowing
+            .put(INDEX_BALANCE_DECIDER_ENABLED_SETTING.getKey(), false)
+            // disable estimated heap decider to avoid spurious shard movements that might cause premature hollowing
+            .put(CLUSTER_ROUTING_ALLOCATION_ESTIMATED_HEAP_THRESHOLD_DECIDER_ENABLED.getKey(), false);
+    }
+
     public void testFieldCapsForHollowShards() throws Exception {
         startMasterOnlyNode();
         var indexNodeSettings = Settings.builder()
@@ -66,7 +81,7 @@ public class StatelessFieldCapabilitiesIT extends AbstractStatelessPluginIntegTe
         startSearchNode();
         ensureStableCluster(3);
 
-        var indexName = randomIdentifier();
+        var indexName = randomIndexName();
         int numberOfShards = randomIntBetween(1, 5);
         assertAcked(prepareCreate(indexName).setMapping(mapping).setSettings(indexSettings(numberOfShards, 1)));
         ensureGreen(indexName);

